### ISSUE TITLE
Docker for Mac fails to run after installation

### ISSUE THREAD

**Issue by katmoody** [docker/for-mac#4023]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: C1C72CC7-E6C5-42F0-8BF4-982078C6B558/20191106161504

### Expected behavior
I'd expect to be able to run Docker after downloading and installing it to my Mac. 

### Actual behavior
Exactly like #4013; Same error and behaves the exact same way. 
<img width="601" alt="Screen Shot 2019-11-06 at 11 19 50 AM" src="https://user-images.githubusercontent.com/5481211/68316420-6e56f900-0087-11ea-8d56-dd2e80956d16.png">

### Information

  - macOS Version:
I'm running MacOS Catalina, version 10.15.1 (19B88)
I downloaded Docker from the website this morning to attempt running it for the first time and it doesn't actually open so I don't have a version number for it. 
Issue is reproducible - I tried installing multiple times to ensure it wasn't a bad download. 

### Diagnostic logs
I uploaded via the tool 

### Steps to reproduce the behavior

1. Download Docker for Mac package.
2. Move Docker.app into /Applications.
3. Start Docker.app.

**Comment by aminueza** [comment]
+1 Same problem

**Comment by mlarcher** [comment]
I upgraded Catalina to 10.5.1 this morning and Docker for mac doesn't start anymore :(

**Comment by aminueza** [comment]
According to this issue #3832, if you go to Launchpad and have it launch Docker it works fine. Directly launching Docker does not seem to work. I don't know enough about "under-the-hood" workings of MacOS but I suspect that it is some type of environmental inheritance when Launchpad fork/exec Docker.

**Comment by katmoody** [comment]
Mine is now working, but I'm not positive why. 
I was doing troubleshooting for a different local installment and ended up finding a symlink for my `/usr/local/bin/` which created a weird redirect loop and wasn't allowing any local environment to run properly. Once I deleted that I was able to get up and running with my other environment and decided to check Docker to just see if the error persisted - no error. 
I'm still not positive that's what changed the issue, but I can now run Docker for local environments and I couldn't before.
